internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -419,3 +421,63 @@ func deployAlerting(ctx context.Context, rr *odhtypes.ReconciliationRequest) err
 
 	return nil
 }
+
+func deployThanosQuerier(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	monitoring, ok := rr.Instance.(*serviceApi.Monitoring)
+	if !ok {
+		return errors.New("instance is not of type *services.Monitoring")
+	}
+
+	if monitoring.Spec.Metrics == nil {
+		rr.Conditions.MarkFalse(
+			status.ConditionThanosQuerierAvailable,
+			conditions.WithReason(status.MetricsNotConfiguredReason),
+			conditions.WithMessage(status.MetricsNotConfiguredMessage),
+		)
```

Comment:
If this is to be always deployed together with metrics stack, I suggest moving this logic to https://github.com/opendatahub-io/opendatahub-operator/blob/3ce28c2d2b29e8f979c093a68e156c9861b968b8/internal/controller/services/monitoring/monitoring_controller_actions.go#L107

Or do we want to deploy monitoring stack even if ThanosQuerier CRD is not present? To me it seems it should be tied together.